lib/Doctrine/ORM/Query/ResultSetMappingBuilder.php

```diff
@@ -154,6 +154,11 @@ protected function addAllClassFields($class, $alias, $columnAliasMap = [])
             }
 
             $this->addFieldResult($alias, $columnAlias, $propertyName);
+
+            $enumType = $classMetadata->getFieldMapping($columnName)['enumType'] ?? null;
```

Comment:
This is the line that is causing the issue. Getting the field mapping seems to have unwanted consequences.

Reply:
Hang on… you're calling a field mapping with a column name here. This is why these tests are failing.

Reply:
```suggestion
            $enumType = $classMetadata->getFieldMapping($propertyName)['enumType'] ?? null;
```

Reply:
indeed, I messed up (and my test was running fine because it was a case where both were equal)